crates/compiler/roc_target/src/lib.rs

```diff
@@ -148,6 +174,51 @@ impl Target {
             Wasm32 => Some("wasm"),
         }
     }
+
+    // file name for a prebuilt host object file
+    // used for legacy linking
+    pub fn prebuilt_static_object(&self) -> String {
+        use Target::*;
+        match self {
+            LinuxX32 | LinuxX64 | LinuxArm64 | MacX64 | MacArm64 | Wasm32 => {
+                format!("{}.o", self)
+            }
+            WinX32 | WinX64 | WinArm64 => {
+                format!("{}.obj", self)
+            }
+        }
+    }
+
+    // file name for a prebuilt host static library file
+    // used for legacy linking
+    pub fn prebuilt_static_library(&self) -> String {
+        use Target::*;
+        match self {
+            LinuxX32 | LinuxX64 | LinuxArm64 | MacX64 | MacArm64 | Wasm32 => {
+                format!("{}.a", self)
+            }
+            WinX32 | WinX64 | WinArm64 => {
+                format!("{}.lib", self)
+            }
+        }
+    }
+
+    // file name for a preprocessed host executable file
+    // used for surgical linking
+    pub fn prebuilt_surgical_host(&self) -> String {
+        format!("{}.rh", self) // short for roc host
+    }
+
+    // file name for a preprocessed host metadata file
+    // used for surgical linking
+    pub fn metadata_file_name(&self) -> String {
+        format!("metadata_{}.rm", self) // short for roc metadata
+    }
+
+    // file name for a stubbed app dynamic library file
+    pub fn stub_app_lib_file_name(&self) -> String {
+        format!("libapp.{}", self.dynamic_library_file_ext())
```

Comment:
On Windows I think it may just be `app.dll`, on unix* systems we need the `lib` prepended